### PR TITLE
feat: set "destination" context for "mongodb" instrumentation

### LINKED ISSUE #1892: "mongodb" module instrumentation is missing "destination" context
Spans emitted by the instrumentation of the [mongodb](https://github.com/mongodb/node-mongodb-native) module do not include the "destination" fields.  The destination fields (specifically `destination.service.resource`) are used by Kibana's APM "Service Map" to display a node for downstream services.

Note that other MongoDB-related Node.js packages, like "mongoose", use the mongodb module, so are also affected.

E.g. here is an Elasticsearch span with "destination":

```
{
  "span": {
    "name": "Elasticsearch: POST /kibana_sample_data_logs/_search",
    "type": "db",
    "id": "891b874cc027aac5",
    "transaction_id": "8cc190e0bf8c117f",
    "parent_id": "8cc190e0bf8c117f",
    "trace_id": "a2ac5d4ddd06d0ea19ddd2fd8d8cd681",
    "subtype": "elasticsearch",
    "action": "request",
    "timestamp": 1606331068870301,
    "duration": 4.103,
    "context": {
      "db": {
        "type": "elasticsearch",
        "statement": "{\"size\":1,\"query\":{\"match_all\":{}}}"
      },
      "destination": {
        "service": {
          "name": "elasticsearch",
          "resource": "elasticsearch",
          "type": "db"
        },
        "address": "localhost",
        "port": 9200
      }
    },
    "sync": false,
    "outcome": "unknown"
  }
}
```

### PR DESCRIPTION
**Pull request by trentm**
Fixes #1892

### CODE REVIEW

**Comment by trentm** [comment]
Current status: just a first pass.

TODOs:

- [ ] test case
- [ ] More robust/defensive handling of the `event.address` etc.

**Comment by trentm** [comment]
TAV test run for me:

```
% TAV=mongodb ./node_modules/.bin/tav --compat
Testing compatibility with mongodb:
✔ 3.6.3
✔ 3.6.2
✔ 3.5.11
✔ 3.6.1
✔ 3.6.0
✔ 3.5.10
✔ 3.5.9
✔ 3.5.8
✔ 3.5.7
✔ 3.5.6
✔ 3.5.5
✔ 3.5.4
✔ 3.5.3
✔ 3.5.2
✔ 3.5.1
✔ 3.5.0
✔ 3.4.1
✔ 3.4.0
✔ 3.3.5
✔ 3.3.4
✔ 3.3.3
✔ 3.3.2
✔ 3.3.1
✔ 3.3.0
```

File: lib/instrumentation/modules/mongodb.js [hunk 1]
```diff
@@ -28,6 +44,19 @@ module.exports = (mongodb, agent, { version, enabled }) => {
     const span = agent.startSpan(name, 'db', 'mongodb', 'query')
     if (span) {
       activeSpans.set(event.requestId, span)
+
+      // Per the following code it looks like "<hostname>:<port>" should be
+      // available via the `address` or `connectionId` field.
+      // https://github.com/mongodb/node-mongodb-native/blob/dd356f0ede/lib/core/connection/apm.js#L155-L169
+      const address = event.address || event.connectionId
+      let match
```
Comment (astorm) [inline lib/instrumentation/modules/mongodb.js:52]: Capturing an online conversation here -- both the docs and some spot checking of a port-less mongo connection (`mongoose.connect('mongodb://localhost/myapp', {useNewUrlParser: true});`) indicate that the address/connection ID will be normalized to include a port, even if the client is configured without one.  If somehow a portless address does end up in there, we won't hard fail as `exec` won't match and the conditional will fail.

**Review by astorm — APPROVED** [review]
code and concepts look solid and reflect what we discovered about this in slack today.  Soft approving, presuming that the lint and ci pass.

**Pull request merged**